src/core/app.lua

```diff
@@ -339,7 +339,7 @@ function apply_config_actions (actions)
       configuration.apps[name] = nil
    end
    function ops.start_app (name, class, arg)
-      local app = class:new(arg)
+      local app = class:new(arg, name)
```

Comment:
This is the bit that enables configuring inter process links without the somewhat redundant configuration by using its own name. I liked that you can simply `config.app(c, "SharedLink", Receiver)`, but I agree at a second glance that this might be unnecessairly magic. Alteratively, the synatx would be: `config.app(c, "AppName", Receiver, "SharedLink")`.